### ISSUE TITLE
Text jumping around in footline

### ISSUE THREAD

**Issue by samcarter** [josephwright/beamer#516]
The text in the footline is jumping around on every 10^{n}th frame. 

See the following MWE

```
\documentclass{beamer}

\usetheme{Madrid}       
\date{\today}

\begin{document}


\begin{frame}
\end{frame}

\addtocounter{framenumber}{8}

\begin{frame}
\end{frame}

\addtocounter{framenumber}{89}

\begin{frame}
\end{frame}

\end{document}
```

![screen shot 2018-11-12 at 16 35 39](https://user-images.githubusercontent.com/43832342/48357958-19fa8280-e69a-11e8-9099-34134bce123a.png)

This bug was originally reported at https://tex.stackexchange.com/questions/452850/when-the-number-of-slides-of-my-beamer-is-more-than-10-the-date-is-not-displaye

(I was trying to fix this in https://github.com/josephwright/beamer/pull/513 but obviously you don't like this, so please find some other solution)

**Comment by louisstuart96** [comment]
Perhaps you would like this solution:
```latex
\documentclass{beamer}

\usetheme{Madrid}

\mode
<presentation>
\makeatletter
\setbeamertemplate{footline}%
{%
\leavevmode%
\hbox{%
\begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{author in head/foot}%
  \usebeamerfont{author in head/foot}\insertshortauthor\expandafter\ifblank\expandafter{\beamer@shortinstitute}{}{~~(\insertshortinstitute)}
\end{beamercolorbox}%
\begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{title in head/foot}%
  \usebeamerfont{title in head/foot}\insertshorttitle
\end{beamercolorbox}%
\begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{date in head/foot}%
  \usebeamerfont{date in head/foot}\hspace*{2ex}\insertshortdate{}\hspace{0pt plus 1filll}
  \usebeamercolor[fg]{page number in head/foot}\usebeamerfont{page number in head/foot}\usebeamertemplate{page number in head/foot}\hspace*{2ex} 
\end{beamercolorbox}}%
\vskip0pt%
}
\makeatother
\mode
<all>

\title{Default}
\author{anonymous}

\date{\today}

\begin{document}

\begin{frame}{Test frame}
\titlepage    
\end{frame}

\addtocounter{framenumber}{8}

\begin{frame}{Test frame}
\end{frame}

\addtocounter{framenumber}{89}

\begin{frame}{Test frame}
\end{frame}

\end{document}
```

It's for @josephwright to decide the final design.

**Comment by samcarter** [comment]
@louisstuart96 Your suggestion will fix the problem cleanly, but will significantly alter the layout of all existing presentations - not my decision if this is to invasive.

In case the page number template is empty it might look very strange, with all other boxes in the footline centred...

**Comment by samcarter** [comment]
closed in #684
